From 12a37096203c9c1820bf9ea80f5e4d277674b670 Mon Sep 17 00:00:00 2001
From: elupus <elupus@568bbfeb-2a22-0410-94d2-cc84cf5bfa90>
Date: Tue, 19 Oct 2010 21:07:10 +0000
Subject: [PATCH] fixed: if container doesn't contain timestamps for all dts packets, lavf used wrong duration of packet to fill in the blanks

The issue was discussed here: http://forum.xbmc.org/showthread.php?t=82896

Ps. The DTS parser is really lacking. It doesn't fill in anything, it only slices up the frames. So if this value changes mid stream, we will start calculating this wrong again. I'm pondering if we should just stop trying to fill in the blanks.

git-svn-id: https://xbmc.svn.sourceforge.net/svnroot/xbmc/trunk@34863 568bbfeb-2a22-0410-94d2-cc84cf5bfa90
---
 libavcodec/dca.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libavcodec/dca.c b/libavcodec/dca.c
index 77299df..7168f41 100644
--- a/libavcodec/dca.c
+++ b/libavcodec/dca.c
@@ -1285,6 +1285,7 @@ static int dca_decode_frame(AVCodecContext * avctx,
     //set AVCodec values with parsed data
     avctx->sample_rate = s->sample_rate;
     avctx->bit_rate = s->bit_rate;
+    avctx->frame_size = s->sample_blocks * 32;
 
     for (i = 0; i < (s->sample_blocks / 8); i++) {
         dca_decode_block(s, 0, i);
-- 
1.7.1

